### PR TITLE
Run Cilium Operator in dev vm

### PR DESCRIPTION
**Pull request by nebril**
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7583)

### CODE REVIEW

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
forgot to start `cilium-operator.service`

**Comment by nebril** [comment]
This can be merged after review without running the CI, it touches only dev vm scripts and config

File: contrib/vagrant/start.sh [hunk 1]
```diff
@@ -370,8 +371,10 @@ if [ -n "\${K8S}" ]; then
     done
 fi
 
+systemctl daemon-reload
 service cilium restart
 /home/vagrant/go/src/github.com/cilium/cilium/test/provision/wait-cilium.sh
+echo "trolo"
```
Comment (ianvernon) [inline contrib/vagrant/start.sh]: Leftover? ;)

**Review by ianvernon — APPROVED** [review]

File: contrib/vagrant/start.sh (hunk 1 above)
Comment (nebril) [reply contrib/vagrant/start.sh]: Lol, yeah

**Comment by nebril** [comment]
leftover deleted, let's :shipit:!

**Comment by nebril** [comment]
@ianvernon just FYI: I moved the restart of `cilium-operator` to avoid having additional operators running on worker nodes

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
Running test due is affecting test/runtime here: 
https://github.com/cilium/cilium/blob/master/test/provision/compile.sh#L43

File: contrib/systemd/cilium [hunk 2]
```diff
@@ -23,6 +23,9 @@ PATH=/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/sbin:/sbin:/bin
 # Check more cli options using cilium-agent -h
 
 CILIUM_OPTS=--kvstore consul --kvstore-opt consul.address=127.0.0.1:8500
+CILIUM_OPERATOR_OPTS=--kvstore consul --kvstore-opt consul.address=127.0.0.1:8500 --k8s-kubeconfig-path=/home/vagrant/.kube/config
```
Comment (ianvernon) [inline contrib/systemd/cilium:26]: Is there a reason this is in the `cilium` file instead of a separate file for cilium operator?

Comment (nebril) [reply contrib/systemd/cilium:26]: No particular reason.

**Comment by nebril** [comment]
hit dns flake

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Comment by nebril** [comment]
test-me-please

**Pull request merged**